install/install.py

```diff
@@ -77,28 +78,25 @@ def install_restricted_from_cfg(self, config_file):
             self.install_in_directory(os.path.expanduser(cp.get(section, "directory")), cp.get(section, "command"))
 
     def install(self):
-        # User asked to select distro and distro related stuff is installed
         cp = ConfigParser.ConfigParser({"RootDir":self.RootDir, "Pid":self.pid})
         cp.read(self.distros_cfg)
 
-        # Loop until proper input is received
-        while True:
-            print("")
-            for i in range(0, len(cp.sections())):
-                print("(%d) %s"%(i+1, cp.sections()[i]))
-            print("(0) %s"%("My distro is not listed :("))
-            distro_num = raw_input("Select a number based on your distribution : ")
-            try: # Cheking if valid input is received
-                distro_num = int(distro_num)
-                if distro_num != 0:
-                    self.run_command(cp.get(cp.sections()[int(distro_num)-1], "install"))
-                else:
-                    print("Skipping distro related installation :(")
-                break
-            except ValueError:
-                print('')
-                print("Please enter a valid number")
-                continue
+        #Try get the distro using lsb_release
+        status, result = commands.getstatusoutput("lsb_release -si")
+        #Use the default way if lsb_release is not installed
+        command = "awk '/DISTRIB_ID=/' /etc/*-release | sed 's/DISTRIB_ID=//' | tr '[:upper:]' '[:lower:]'"
+        if status != 0:
+            status, result = commands.getstatusoutput(command)
+        distro_num = 0
+        if "kali" in result.lower():
+            distro_num = 1
+        elif "samurai" in result.lower():
+            distro_num = 2
+
+        if distro_num != 0:
+            self.run_command(cp.get(cp.sections()[int(distro_num)-1], "install"))
+        else:
```

Comment:
If we cannot detect the distro, please prompt the user to select one. There are cases when someones takes a distribution and then add few new sources and customizations and rename it or someone using debian but has all the kali repos. So, if we are unable to detect known distros prompt the user to select one